lib/src/components/NxTabs/__tests__/NxTabs.test.tsx

```diff
@@ -4,104 +4,77 @@
-        </NxTabList>
-      </NxTabs>
-    );
+describe('NxTabs', function() {
+  const minimumProps = {
+    onTabSelect: () => {},
+    children: (
+      <NxTabList>
+        <NxTab>Tabs 0</NxTab>
+      </NxTabList>
+    )
+  };
 
-    expect(firstTabs).not.toHaveProp('id', secondTabs.prop('id'));
-  });
+  const quickRender = rtlRender<NxTabsProps>(NxTabs, minimumProps);
+  const renderEl = rtlRenderElement<NxTabsProps>(NxTabs, minimumProps);
 
-  it('renders with the correct classnames', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1} className="test">
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('merges any passed in className', function() {
+    const tabsWithAddedClassName = renderEl({ className: 'foo' });
+    const tabs = renderEl()!;
 
-    expect(component).toHaveClassName('nx-tabs test');
+    expect(tabsWithAddedClassName).toHaveClass('foo');
+
+    for (const cls of Array.from(tabs.classList)) {
+      expect(tabsWithAddedClassName).toHaveClass(cls);
+    }
   });
 
-  it('renders a generated id', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1}>
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('renders with a unique id each time', function() {
+    const firstTabs = renderEl({ activeTab: 0 })!;
+    const secondTabs = renderEl({ activeTab: 0 })!;
 
-    expect(component.prop('id')).toMatch(/^nx-tabs-\d+$/);
+    expect(firstTabs.id).not.toEqual(secondTabs.id);
   });
 
-  it('renders using as specific id', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1} id="my-tabs">
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
-
-    expect(component.prop('id')).toMatch('my-tabs');
+  it('renders a generated id', function() {
+    const component = renderEl({ activeTab: -1 })!;
+    expect(component.id).toMatch(/^nx-tabs-\d+$/);
   });
 
-  it('renders no tab contents when none are active', function () {
-    const component = mount(
-      <NxTabs onTabSelect={() => {}} activeTab={-1}>
+  it('renders using as specific id', function() {
+    const { container, getByRole } = render(
+      <NxTabs activeTab={0} id="my-tabs" onTabSelect={() => {}}>
         <NxTabList>
           <NxTab>Tab 0</NxTab>
         </NxTabList>
         <NxTabPanel>Content 0</NxTabPanel>
       </NxTabs>
     );
 
-    expect(component.find(NxTab)).not.toHaveClassName('active');
-    expect(component.find(NxTabPanel)).toBeEmptyRender();
+    expect(container.firstElementChild!.id).toMatch('my-tabs');
+    expect(getByRole('tab')!.id).toMatch('my-tabs');
+    expect(getByRole('tabpanel')!.id).toMatch('my-tabs');
```

Comment:
Maybe make sure that each of these doesn't exactly match `my-tabs` nor exactly match each other

Reply:
> Maybe make sure that each of these doesn't exactly match `my-tabs` nor exactly match each other

yup: updated here: f42f83d3237e208639e1aabf8f86c0ab7c613e06

Reply:
> yup: updated here: https://github.com/sonatype/sonatype-react-shared-components/commit/f42f83d3237e208639e1aabf8f86c0ab7c613e06

I misspoke slightly above.  The id of the `nx-tabs` itself _should_ exactly match "my-tabs", and this test should make sure of that.

Reply:
> > yup: updated here: [f42f83d](https://github.com/sonatype/sonatype-react-shared-components/commit/f42f83d3237e208639e1aabf8f86c0ab7c613e06)
> 
> I misspoke slightly above. The id of the `nx-tabs` itself _should_ exactly match "my-tabs", and this test should make sure of that.

Ops, fixed here (Also updated stateful test): de8fa9f56f2a050b7784c21f760f02ec84bee096